be/src/olap/cumulative_compaction_time_series_policy.cpp

```diff
@@ -288,7 +349,40 @@ void TimeSeriesCumulativeCompactionPolicy::update_cumulative_point(
         return;
     }
 
+    if (tablet->tablet_meta()->time_series_compaction_level_threshold() == 2 &&
+        output_rowset->rowset_meta()->compaction_level() < 2) {
+        return;
+    }
+
     tablet->set_cumulative_layer_point(output_rowset->end_version() + 1);
 }
 
+void TimeSeriesCumulativeCompactionPolicy::update_compaction_level(
+        Tablet* tablet, const std::vector<RowsetSharedPtr>& input_rowsets,
+        RowsetSharedPtr output_rowset) {
+    if (tablet->tablet_state() != TABLET_RUNNING || output_rowset->num_segments() == 0) {
+        return;
+    }
+
+    int64_t first_level = 0;
+    for (size_t i = 0; i < input_rowsets.size(); i++) {
+        int64_t cur_level = input_rowsets[i]->rowset_meta()->compaction_level();
+        if (i == 0) {
+            first_level = cur_level;
+        } else {
+            if (first_level != cur_level) {
+                LOG(ERROR) << "Failed to check compaction level, first_level: " << first_level
```

Comment:
What's the effect of LOG(ERROR)? If it just add a log and continue, it's not enough.